### PR TITLE
Add git notes functionalities: add, append, list, show.

### PR DESCRIPTION
**Pull request by civa86**

### CODE REVIEW

File: src/clj_jgit/porcelain.clj [hunk 1]
```diff
@@ -795,3 +797,61 @@
...
+           (map #(String. (.getBytes (.open repository (.getData %))) (StandardCharsets/UTF_8)))
+           (map #(str/split % #"\n"))
+           (first))))
+  ([^Git repo]
+    (git-notes-show repo "commits")))
+
+(defn get-head-commit-object "Return HEAD RevCommit instance" [^Git repo ]
+  (let [repository (-> repo .getRepository)
+        head-id (-> repository (.resolve "HEAD"))]
+    (-> (RevWalk. repository) (.parseCommit head-id))))
```
[the remark this answers is not in the document]
Comment (civa86) [reply src/clj_jgit/porcelain.clj]: Totally agree with your review. I pushed changes.

**Comment by edannenberg** [comment]
Thanks! New clojars snapshot is up.

**Comment by edannenberg** [comment]
Thanks! New clojars snapshot is up.

**Comment by edannenberg** [comment]
Thanks! New clojars snapshot is up.

**Pull request closed without merging**